### ISSUE TITLE
Some questions about the code

### ISSUE THREAD

**Issue by codehangyan** [ravikothari510/crossattention_radar_detector#3]
Dear Professor Ravi Kothari,
I am currently trying to run your GitHub, but have run into some issues that I was hoping you could help provide some guidance on:
1.	Runtime errors - Input type double and bias type (float) should be the same, and I had to modify the code in dataset_process to:
Changed
np.save(os.path.join(seq_dir, 'ra_map', str(track[-1]).zfill(6) + '.npy'), ra_map)
np.save(os.path.join(seq_dir, 'rd_map', str(track[-1]).zfill(6) + '.npy'), rd_map)
np.save(os.path.join(seq_dir, 'ad_map', str(track[-1]).zfill(6) + '.npy'), ad_map)
to
np.save(os.path.join(seq_dir, 'ra_map', str(track[-1]).zfill(6) + '.npy'), np.float32(ra_map)) np.save(os.path.join(seq_dir, 'rd_map', str(track[-1]).zfill(6) + '.npy'), np.float32(rd_map)) np.save(os.path.join(seq_dir, 'ad_map', str(track[-1]).zfill(6) + '.npy'), np.float32(ad_map))

I also modified main.py model initialization to:
model = get_model(args=args).to(args.device)

2.	Runtime errors - Boolean values:
I encountered runtime errors related to boolean values in transform.py and val_utils.py. I made the following adjustments:
In transform.py: Changed if r_map != 0: to if torch.any(r_map != 0):
In val_utils.py: Changed if pred_c != 0: to if (pred_c != 0).any(): In validation function: Changed if pr_o != 0.any: to if (pr_o != 0).any():
Despite these changes, when executing the association function in val_utils.py, specifically when the code execution reaches:
if epoch % 5 == 0: _= evaluation(loader=val_loader, model=model, args=args)
When attempting to run the evaluation process after completing 10 training epochs, executing the association function in val_utils.py becomes extremely slow to the point where it cannot finish running even after several days.
I would be most grateful for any guidance or suggestions you can provide in resolving this performance issue. Your expertise and insights are critical to ensuring I can successfully reproduce your work.
Thank you very much in advance for your time and assistance. I look forward to your reply.
Best regards,

My hardware is an NVIDIA GeForce RTX 4090 GPU, and I processed the CARADA data downloaded from the official website using the code you provided. The torch version I am using is 1.13.1. The training is as follows:
python train.py --frame 5 --oren 1 
Epoch 0: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 266/266 [00:44<00:00,  5.99it/s, loss=5.15e+3]
Training time for one epoch: 44.37665629386902 seconds
Mean Training loss : nan
mean MSE : 5790.5341796875 mean focal : 6701.59814453125 mean center loss : 2221.9716796875 mean orent loss : 21.033824920654297
Epoch 0 time: 47.254741191864014 seconds
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 57/57 [20:40<00:00, 21.77s/it]
0 AP: 0.14795942839562234 P:0.023509387798521356 R: 0.8051001821493625 GT: 549
1 AP: 0.029948588364357386 P:0.030683403068340307 R: 0.08979591836734693 GT: 245
2 AP: 0.019054039818869817 P:0.0326530612244898 R: 0.049586776859504134 GT: 484
Total evaluation time: 1241.0139014720917 seconds
Epoch 1: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 266/266 [00:43<00:00,  6.16it/s, loss=4.84e+3]
Training time for one epoch: 43.20922660827637 seconds
Mean Training loss : nan
mean MSE : 3757.7412109375 mean focal : 5668.9609375 mean center loss : 2038.0849609375 mean orent loss : 21.48600196838379
Epoch 1 time: 48.63705825805664 seconds
Epoch 2: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 266/266 [00:42<00:00,  6.25it/s, loss=4.08e+3]
Training time for one epoch: 42.52754998207092 seconds
Mean Training loss : nan
mean MSE : 5378.62890625 mean focal : 5079.19189453125 mean center loss : 1954.149169921875 mean orent loss : 21.87099266052246
Epoch 2 time: 46.99090766906738 seconds
Epoch 3: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 266/266 [00:42<00:00,  6.30it/s, loss=5.11e+3]
Training time for one epoch: 42.223567724227905 seconds
Mean Training loss : nan
mean MSE : 4063.818359375 mean focal : 5510.89013671875 mean center loss : 1930.176513671875 mean orent loss : 21.361854553222656
Epoch 3 time: 45.727668046951294 seconds
Epoch 4: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 266/266 [00:43<00:00,  6.17it/s, loss=4.83e+3]
Training time for one epoch: 43.12650513648987 seconds
Mean Training loss : nan
mean MSE : 2666.145751953125 mean focal : 4371.3642578125 mean center loss : 1900.803955078125 mean orent loss : 16.73015785217285
Epoch 4 time: 47.650901556015015 seconds
Epoch 5: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 266/266 [00:43<00:00,  6.14it/s, loss=3.5e+3]
Training time for one epoch: 43.34084177017212 seconds
Mean Training loss : nan
mean MSE : 2539.724853515625 mean focal : 7057.11328125 mean center loss : 1890.0465087890625 mean orent loss : 15.169075012207031
Epoch 5 time: 46.95159077644348 seconds
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 57/57 [02:23<00:00,  2.52s/it]
0 AP: 0.299475001512758 P:0.0533625730994152 R: 0.6648451730418944 GT: 549
1 AP: 0.01708398177785933 P:0.033707865168539325 R: 0.036734693877551024 GT: 245
2 AP: 0.8363245611576101 P:0.36544585987261147 R: 0.9483471074380165 GT: 484
Total evaluation time: 143.84979462623596 seconds
Epoch 6: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 266/266 [00:41<00:00,  6.34it/s, loss=4.07e+3]
Training time for one epoch: 41.94576692581177 seconds
Mean Training loss : nan
mean MSE : 49044.921875 mean focal : 19781.775390625 mean center loss : 2024.5999755859375 mean orent loss : 17.262584686279297
Epoch 6 time: 45.04865312576294 seconds
Epoch 7: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 266/266 [00:43<00:00,  6.15it/s, loss=1.39e+4]
Training time for one epoch: 43.286442041397095 seconds
Mean Training loss : nan
mean MSE : 17538.29296875 mean focal : 16458.380859375 mean center loss : 2363.576416015625 mean orent loss : 21.564939498901367
Epoch 7 time: 46.37185454368591 seconds
Epoch 8: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 266/266 [00:42<00:00,  6.22it/s, loss=1.95e+4]
Training time for one epoch: 42.798298835754395 seconds
Mean Training loss : nan
mean MSE : 778027.0 mean focal : 2242658.0 mean center loss : 2638.1787109375 mean orent loss : 19.774333953857422
Epoch 8 time: 45.87578463554382 seconds
Epoch 9: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 266/266 [00:44<00:00,  6.01it/s, loss=9.77e+4]
Training time for one epoch: 44.25476884841919 seconds
Mean Training loss : nan
mean MSE : 225068.8125 mean focal : 135873.59375 mean center loss : 2395.76953125 mean orent loss : 19.627391815185547
Epoch 9 time: 47.392253160476685 seconds
Epoch 10: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 266/266 [00:43<00:00,  6.13it/s, loss=1.35e+5]
Training time for one epoch: 43.42087149620056 seconds
Mean Training loss : nan
mean MSE : 275120.40625 mean focal : 188934.765625 mean center loss : 2357.1630859375 mean orent loss : 20.712303161621094
Epoch 00011: reducing learning rate of group 0 to 1.0000e-05.
Epoch 10 time: 46.54237365722656 seconds
  0%|                                                                                                                                                                                                                                                                    | 0/57 [00:00<?, ?it/s]
